vue/sbc-common-components/src/components/BrowserVersionAlert.vue

```diff
@@ -0,0 +1,110 @@
+<template>
+  <sbc-system-banner
+    v-if="browserSupported === false"
+    v-bind:show="browserSupported === false"
+    type="warning"
+    v-bind:message="alertMessage"
+  ></sbc-system-banner>
+</template>
+
+<script lang='ts'>
+import { Component, Vue } from 'vue-property-decorator'
+import SbcSystemBanner from './SbcSystemBanner.vue'
+
+declare module 'vuex' {
+  interface Store<S> {
+    hasModule(_: string[]): boolean
+  }
+}
+
+@Component({
+  components: {
+    SbcSystemBanner
+  },
+  beforeCreate () {
+    this.$store.isModuleRegistered = function (aPath: string[]) {
+      let m = (this as any)._modules.root
+      return aPath.every((p) => {
+        m = m._children[p]
+        return m
+      })
+    }
+  }
+})
+export default class BrowserVersionAlert extends Vue {
+  private alertMessage: string = ''
+  private browserSupported: boolean = true
+
+  private getBrowser (): { name: string; version: number } {
+    let ua = navigator.userAgent
+    let tem: string[]
+    let M = ua.match(/(opera|chrome|safari|firefox|msie|trident(?=\/))\/?\s*(\d+)/i) || []
+
+    if (/trident/i.test(M[1])) {
+      tem = /\brv[ :]+(\d+)/g.exec(ua) || []
+      return { name: 'IE', version: parseInt(tem[1] || '') }
+    }
+    if (M[1] === 'Chrome') {
+      tem = ua.match(/\bOPR\/(\d+)/)
+      if (tem != null) { return { name: 'Opera', version: parseInt(tem[1]) } }
+    }
+
+    // MS Edge Html
+    if (window.navigator.userAgent.indexOf('Edge') > -1) {
+      // eslint-disable-next-line no-useless-escape
+      tem = ua.match(/\Edge\/(\d+)/)
+
+      if (tem != null) {
+        return { name: 'Edge', version: parseInt(tem[1]) }
+      }
+    }
+
+    // MS Edge Chromium
+    if (window.navigator.userAgent.indexOf('Edg') > -1) {
+      // eslint-disable-next-line no-useless-escape
+      tem = ua.match(/\Edg\/(\d+)/)
+
+      if (tem != null) {
+        return { name: 'Edge', version: parseInt(tem[1]) }
+      }
+    }
+
+    M = M[2] ? [M[1], M[2]] : [navigator.appName, navigator.appVersion, '-?']
+
+    if ((tem = ua.match(/version\/(\d+)/i)) != null) {
+      M.splice(1, 1, tem[1])
+    }
+    return {
+      name: M[0],
+      version: +M[1]
+    }
+  }
+
+  private isSupported (browser: { name: string; version: number }): boolean {
+    var supported = true
+
+    if ((browser.name === 'MSIE' || browser.name === 'IE') && browser.version <= 11) {
+      supported = false
+    }
+    return supported
+  }
+
+  private async mounted () {
+    this.browserSupported = this.isSupported(this.getBrowser())
+    if (!this.browserSupported) {
+      // render unsupported message
+      this.alertMessage = 'Unsupported Browser! We only support the recent versions of major browsers like Chrome, Firefox, Safari, and Edge.'
+      console.log(this.alertMessage)
```

Comment:
This line may need a "lint disable" comment. (Depends on lint configuration whether console.xxx() generated a warning/error.) Even if not needed, it's probably good practise to add such a comment anyway.

Reply:
Removed.